Maple2.Server.Game/Model/Field/Actor/ActorState/AnimationState.cs

```diff
@@ -0,0 +1,233 @@
+﻿using Maple2.Model.Metadata;
+using Maple2.Server.Core.Packets;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static IronPython.Modules._ast;
+
+namespace Maple2.Server.Game.Model.Field.Actor.ActorState;
+
+public class AnimationState {
+    // TODO: maybe make this a ping based variable?
+    private const long ClientGraceTimeTick = 500; // max time to allow client to go past loop & sequence end
+    private readonly IActor actor;
+
+    public enum Type {
+        Misc,
+        Move,
+        Skill
+    }
+
+    private struct LoopData {
+        public float start;
+        public float end;
+
+        public LoopData(float start, float end) {
+            this.start = start;
+            this.end = end;
+        }
+    }
+
+    public struct TickPair {
+        public long server;
+        public long client;
+
+        // mobs
+        public TickPair(long server) {
+            this.server = server;
+            this.client = server;
+        }
+
+        // players
+        public TickPair(long server, long client) {
+            this.server = server;
+            this.client = client;
+        }
+    }
```

Comment:
I originally put it in thinking there were client ticks sent on skill packets, but I didn't see any when I was hooking it up. I'm thinking of removing the tick pair entirely since it doesn't seem necessary in the case players use the animation state for.